### PR TITLE
Turn on F401: Unused import warning.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#18598 Turn on F401: Unused import warning.**

This was requested by someone at Facebook; this lint is turned
on for Facebook by default.  "Sure, why not."

I had to noqa a number of imports in __init__.  Hypothetically
we're supposed to use __all__ in this case, but I was too lazy
to fix it.  Left for future work.

Be careful!  flake8-2 and flake8-3 behave differently with
respect to import resolution for # type: comments.  flake8-3 will
report an import unused; flake8-2 will not.  For now, I just
noqa'd all these sites.

All the changes were done by hand.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D14687478](https://our.internmc.facebook.com/intern/diff/D14687478)

### CODE REVIEW

**Review by soumith — APPROVED** [review]
i'm a bit sad that you added the ugly # noqa comments everywhere, but whatever rocks your boat

**Comment by ezyang** [comment]
The `__init__` ones are fixable, we just have to add `__all__` definitions to the modules. But it's nontrivial sometimes because there are `*` imports in some cases.

File: test/test_module/future_div.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from __future__ import division
-import torch
+import torch  # noqa: F401
```
Comment (ezyang) [inline test/test_module/future_div.py]: @zou3519 I assumed that the `import torch` here was intentional

File: torch/hub.py [hunk 2]
```diff
@@ -2,18 +2,14 @@
...
 
 if sys.version_info[0] == 2:
     from urlparse import urlparse
     from urllib2 import urlopen  # noqa f811
 else:
     from urllib.request import urlopen
-    from urllib.parse import urlparse
-
-import torch
-import torch.utils.model_zoo as model_zoo
```
Comment (ezyang) [inline torch/hub.py]: @ailzhang Was this import intentional (to reexpose it as `torch.hub.model_zoo`?) If so, I can add it back.

Comment (ailzhang) [reply torch/hub.py]: Yea that was one confusion we had about model_zoo and hub I think. I have a meeting with @soumith in 2hrs. I will bring this up for discussion during the meeting and get back to you here. Thanks!

File: test/test_module/future_div.py (hunk 1 above)
Comment (zou3519) [reply test/test_module/future_div.py]: yes, although nothing bad will happen if we delete it. This file is supposed to emulate some code that a user would write

**Comment by eellison** [comment]
Somewhat related to this change: maybe update the Contributing.md from 
`pip install flake8-mypy` to `pip install flake8 flake8-mypy flake8-comprehensions flake8-pyi mccabe pycodestyle pyflakes` ?

**Comment by ezyang** [comment]
@eellison Yes good catch, I'll do it.

**Comment by eellison** [comment]
@ezyang i think you had a bad merge

**Comment by ezyang** [comment]
Ugh, no the merge is fine but GitHub didn't figure it out correctly. Irritating.

File: torch/hub.py (hunk 2 above)
Comment (ailzhang) [reply torch/hub.py]: @ezyang Feel free to remove them for now. I will make a followup PR to redirect `torch.utils.model_zoo` properly to `torch.hub` with corresponding doc changes. No need to block this PR from merging. Thanks!

**Pull request closed without merging**